tests/test_ringsSkyMap.py

```diff
@@ -77,6 +89,95 @@ def testDm7770(self):
             self.skymap.findAllTracts(coord)
         self.skymap.findTractPatchList(coordList)
 
+    def testDm14809(self):
+        """Test that DM-14809 has been fixed
+
+        The observed behaviour was:
+
+            skyMap.findTract(skyMap[9712].getCtrCoord()).getId() != 9712
+
+        and
+
+            skyMap[1].getCtrCoord() == skyMap[11].getCtrCoord()
+
+        In order to be thorough, we generalise these over the entire skymap.
+        """
+        # Check that the tract ID is that used to index the skymap
+        expect = [tract.getId() for tract in self.skymap]
+        got = [self.skymap.findTract(tract.getCtrCoord()).getId() for tract in self.skymap]
+        self.assertListEqual(got, expect)
+
+        # Check that the tract found for central coordinate of a tract is that tract
+        centers = set([(coord.getRa().asRadians(), coord.getDec().asRadians()) for
+                       coord in (tract.getCtrCoord() for tract in self.skymap)])
+        self.assertEqual(len(centers), len(self.skymap))
+
+        # Check that the first tract in the last ring exists
+        coord = self.getFirstTractLastRingCoord()
+        tract = self.skymap.findTract(coord)
+        self.assertTrue(tract.contains(coord))
```

Comment:
Please eliminate the code duplication with the new methods in SkyMapTestCase.

Preferably please have this test case and the next one inherit from SkyMapTestCase (but the next one, for version 0 maps, would need to override one or both of the base class test methods).

Reply:
This test case is for testing the specific instance of the skymap used by HSC. It doesn't need the more general tests contained in the `SkyMapTestCase`. In fact, since it is has many more tracts than the ones usually used with `SkyMapTestCase`, inheriting would increase the test runtime substantially without providing much (if any) additional safety.